### PR TITLE
🧹 Restructure CF functions code into smaller files

### PR DESCRIPTION
**Pull request by danyx23**
This PR restructures the by now pretty large grapherRenderer.ts file into a few smaller files that roughly group similar functionality.

### CODE REVIEW

**Comment by danyx23** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/4054?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3603** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3603?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4014** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/4014?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4054** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/4054?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4051** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/4051?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3613** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3613?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3884** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3884?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3922](https://github.com/owid/owid-grapher/pull/3922) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3922?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3880** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3880?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3879** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3879?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3895** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3895?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3870** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3870?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3871](https://github.com/owid/owid-grapher/pull/3871) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3871?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3867** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3867?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3865](https://github.com/owid/owid-grapher/pull/3865) <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3865?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3945** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3945?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danyx23 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by marcelgerber — APPROVED** [review]
Lords work! Thank you 🙏

**Comment by danyx23** [comment]
Hi @marcelgerber! I just polished and rebased this - I think it's good to merge from my side. Can you give it a last look to see if you are happy with the changes? There is one changeset with everything since your first review.

**Review by ikesau — APPROVED** [review]
Always difficult to review these sorts of PRs with high confidence that I caught everything, but I tested each route locally with 2 slugs/uuids each and they all worked (except the human trafficking one) so I think we should be good?

File: functions/_common/metadataTools.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    )
+
+    return grapher.inputTable.getColumns(colsToGet)
+}
+export function assembleMetadata(
+    grapher: Grapher,
+    searchParams: URLSearchParams
+) {
+    const useShortNames = searchParams.get("useColumnShortNames") === "true"
+    console.log("useShortNames", useShortNames)
```
[the remark this answers is not in the document]
Comment (danyx23) [reply functions/_common/metadataTools.ts]: Good point, removed this one and another one.

File: functions/_common/downloadFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    const fullMetadata = assembleMetadata(
+        grapher,
+        searchParams ?? new URLSearchParams("")
+    )
+
+    return Response.json(fullMetadata)
+}
+
+export async function fetchZipForGrapher(
```
[the remark this answers is not in the document]
Comment (danyx23) [reply functions/_common/downloadFunctions.ts:34]: A good catch. I fixed this (wrong handling of origins in one place).

**Pull request merged**